contracts/netmap/contract.go

```diff
@@ -278,6 +302,37 @@ func AddPeer(nodeInfo []byte) {
 	}
 }
 
+// AddNode adds a new node into the candidate list for the next epoch. Node
+// must have Online state to be considered and the request must be signed by
+// both the node and Alphabet. AddNode event is emitted upon success.
+func AddNode(n Node2) {
+	ctx := storage.GetContext()
+
+	if n.State != nodestate.Online {
+		panic("can't add non-online node")
+	}
+
+	common.CheckWitness(n.Key)
+	common.CheckAlphabetWitness(common.AlphabetAddress())
+
+	var key = append([]byte(node2CandidatePrefix), n.Key...)
+	storage.Put(ctx, key, std.Serialize(n))
+	runtime.Notify("AddNode", n.Key, n.Addresses, n.Attributes)
```

Comment:
Good question. Initially I've used key only, but this makes notification somewhat more useful. Can be discussed. 1K seems to be enough for this data at the same time.

Reply:
> 1K seems to be enough for this data at the same time.

To me it also looks enough for real-life usecases, if you don't intentionally try to break the node addition process using some unrealistically long attributes. So probably we should just document this system constraint.

Reply:
Test execution will fail -> no transaction created -> node will whine about something in log

Reply:
node is not in the netmap and all we know is there will be some log in the IR's output? i think this should be at least mentioned somewhere to be able to put SN's admin to the info that it will never find himself

Reply:
Why IR? _SN_ couldn't create a transaction.

Reply:
so you think SN waits for notary request inclusion? it just sends it and waits for a better future. currently, all the problem code is on the IR's side

OR you want to say that it would fail on the creation step? SN sends notary request and it fails immediately on `runtime.Notify` even if there is no notification yet?

Reply:
>  it would fail at creation step? 

Exactly this. SN does test runs to estimate GAS at least. This will fail.